plugin/cache.go

```diff
@@ -240,6 +236,10 @@ func cachedOp(ctx context.Context, opName string, entry Entry, ttl time.Duration
 		}
 	}
 
+	if ttl < 0 {
```

Comment:
Had to do this because of the `apifs` package, where those entries do not have parents (that's also why I removed the g.id() == "" panic in `CachedList`).

Times when an empty ID would be bad is if someone tries to use a `CachedOp` (which the apifs entries don't), or if someone tries to use `plugin.ID`. I can extend `plugin.ID` to panic on an empty entry ID.

Reply:
In practice the extra branches for failure checking are pretty much always worth it.